stack/manager.go

```diff
@@ -573,7 +575,7 @@ func (m *Manager) tfModuleChanged(
 }
 
 func (m *Manager) tgModuleChanged(
-	stack *config.Stack, tgMod *tg.Module, gitBaseRef string, changedFiles []string, stackSet map[project.Path]Entry, tgModuleMap map[project.Path]*tg.Module,
+	stack *config.Stack, tgMod *tg.Module, gitBaseRef string, stackSet map[project.Path]Entry, tgModuleMap map[project.Path]*tg.Module,
```

Comment:
An idea:
`tgModuleChanged` now relies on the cache being populated, which could lead to bugs in the future.

I think it would be nicer to have a function

```
func (m) cachedChangedFiles(hostDir, gitBaseRef) {
if m.cache.changedFiles == nil {
		var err error

		m.cache.changedFiles, err = m.listChangedFiles(m.root.HostDir(), gitBaseRef)
		if err != nil {
			return nil, errors.E(errListChanged, err)
		}
	}
}
```
that is invoked by the caller of `tgModuleChanged`. In general, make it so it doesn't matter who invoked the computation first.

But only if it's possible, i.e. I don't know if all the callers have the arguments available.

Reply:
good idea, thanks!

Reply:
will apply in later PR.